tests/parser/functions/test_concat.py

```diff
@@ -121,6 +123,56 @@ def small_output(a: String[5], b: String[4]) -> String[9]:
     assert c.small_output("", "") == ""
 
 
+def test_small_bytes(get_contract_with_gas_estimation):
+    # TODO maybe use parametrization or hypothesis for the examples
+    code = """
+@external
+def small_bytes1(a: bytes1, b: Bytes[2]) -> Bytes[3]:
+    return concat(a, b)
+
+@external
+def small_bytes2(a: Bytes[1], b: bytes2) -> Bytes[3]:
+    return concat(a, b)
+
+@external
+def small_bytes3(a: bytes4, b: bytes32) -> Bytes[36]:
+    return concat(a, b)
+
+@external
+def small_bytes4(a: bytes8, b: Bytes[32], c: bytes8) -> Bytes[48]:
+    return concat(a, b, c)
+    """
+    contract = get_contract_with_gas_estimation(code)
+
+    def bytes_for_len(n):
+        # note: python 3.9 has `randbytes` for this.
+        return random.randrange(2 ** (8 * n)).to_bytes(n, "big")
```

Comment:
Can't you test values for bytes that are edge cases of behavior? My problem with introducing non-determinism into the test suite like this is that there is not an easy way to replicate a failure by setting the seed like you can in hypothesis.

Reply:
how about if we set and report the random seed used? https://github.com/charles-cooper/vyper/commit/77c2b1d11f1582921d82f94c058d360d74ca0b90

Reply:
@fubuloubu resolved in https://github.com/vyperlang/vyper/pull/2764/commits/0fa645c25f3f56549b0660d013d9debb79c7780c

Reply:
actually -- see https://github.com/vyperlang/vyper/pull/2764/commits/e8ad0508740c03645e7fcb7c4fd516cb2ffdb6e7